azure-servicebus/src/test/java/com/microsoft/azure/servicebus/TestCommons.java

```diff
@@ -214,20 +270,36 @@ public static void testBasicReceiveAndRenewLockBatch(IMessageSender sender, Stri
 		}		
 	}
 		
-	public static void testBasicReceiveBatchAndComplete(IMessageSender sender, String sessionId, IMessageReceiver receiver) throws InterruptedException, ServiceBusException, ExecutionException
+	public static void testBasicReceiveBatchAndComplete(IMessageSender sender, String sessionId, IMessageReceiver receiver, boolean isEntityPartitioned) throws InterruptedException, ServiceBusException, ExecutionException
 	{
 		int numMessages = 10;		
-		List<Message> messages = new ArrayList<Message>();
-		for(int i=0; i<numMessages; i++)
+		if(isEntityPartitioned)
 		{
-			Message message = new Message("AMQP message");
-			if(sessionId != null)
+			for(int i=0; i<numMessages; i++)
 			{
-				message.setSessionId(sessionId);
+				Message message = new Message("AMQP message");
+				if(sessionId != null)
+				{
+					message.setSessionId(sessionId);
+				}
+				sender.send(message);
 			}
-			messages.add(message);
 		}
-		sender.sendBatch(messages);
+		else
+		{
+			// Keep this batch send part as this test sendBatch API call
+			List<Message> messages = new ArrayList<Message>();
+			for(int i=0; i<numMessages; i++)
+			{
+				Message message = new Message("AMQP message");
+				if(sessionId != null)
+				{
+					message.setSessionId(sessionId);
+				}
+				messages.add(message);
+			}
+			sender.sendBatch(messages);
+		}
```

Comment:
This could have been a util method here instead

Reply:
It repeats only twice, so didn't take it out to a separate util method.

---
In reply to: [238019126](https://github.com/Azure/azure-service-bus-java/pull/309#discussion_r238019126) [](ancestors = 238019126)